src/Compilers/CSharp/Portable/Binder/Semantics/Operators/UnaryOperatorOverloadResolution.cs

```diff
@@ -319,6 +320,36 @@ private bool GetUserDefinedOperators(UnaryOperatorKind kind, BoundExpression ope
                 }
             }
 
+            // Look in base interfaces, or effective interfaces for type parameters  
+            if (!hadApplicableCandidates)
+            {
+                ImmutableArray<NamedTypeSymbol> interfaces = default;
+                if (type0.IsInterfaceType())
+                {
+                    interfaces = type0.AllInterfacesWithDefinitionUseSiteDiagnostics(ref useSiteDiagnostics);
+                }
+                else if (type0.IsTypeParameter())
+                {
+                    interfaces = ((TypeParameterSymbol)type0).AllEffectiveInterfacesWithDefinitionUseSiteDiagnostics(ref useSiteDiagnostics);
+                }
+
+                if (!interfaces.IsDefaultOrEmpty)
+                {
+                    operators.Clear();
+                    results.Clear();
+
+                    foreach (NamedTypeSymbol @interface in interfaces)
+                    {
+                        GetUserDefinedUnaryOperatorsFromType(@interface, kind, name, operators);
```

Comment:
For simplicity, the overload resolution is going to do the right thing any way and each interface can define at most one unary operator of each kind. Tracking inheritance relationship in interfaces is not that trivial due to multiple inheritance. Effectively, the end result is going to be the same.